tests/integration/test_where.py

```diff
@@ -65,17 +157,22 @@ def test_argwhere(input):
     assert np.array_equal(np.argwhere(anp), num.argwhere(a))
 
 
-@pytest.mark.skip
-def test_extract():
-    cnp = np.array(
-        [1, 54, 4, 4, 0, 45, 5, 58, 0, 9, 0, 4, 0, 0, 0, 5, 0, 1]
-    ).reshape(
-        (6, 3)
-    )  # noqa E501
-    c = num.array(cnp)
-    bnp = np.random.randn(6, 3)
-    b = num.array(bnp)
-    assert num.array_equal(num.extract(c, b), np.extract(cnp, bnp))
+@pytest.mark.xfail
+def test_argwhere_none():
+    # In Numpy, it pass and returns []
+    # In cuNumeric, it raises AttributeError:
+    # 'NoneType' object has no attribute '_thunk'
+    assert np.array_equal(np.argwhere(None), num.argwhere(None))
```

Comment:
@manopapad so what do you want to happen in cases like this, e.g. ? The `np.where` docstring just states:
```
condition : array_like, bool
```
and I think I'd expect None to be covered by one or both of those (since None is Falsey)

Reply:
This particular test is talking about `np.argwhere`, so I'll discuss this on this thread, and start separate threads for the others.

mypy reveal_type reveals the following type for `np.argwhere` (formatting mine):

```
def (a: ArrayLike) -> numpy.ndarray[Any, numpy.dtype[numpy.signedinteger[Any]]]
```

where:

```
ArrayLike = Union[
  numpy._typing._array_like._SupportsArray[numpy.dtype[Any]],
  numpy._typing._nested_sequence._NestedSequence[numpy._typing._array_like._SupportsArray[numpy.dtype[Any]]],
  builtins.bool,
  builtins.int,
  builtins.float,
  builtins.complex,
  builtins.str,
  builtins.bytes,
  numpy._typing._nested_sequence._NestedSequence[Union[builtins.bool, builtins.int, builtins.float, builtins.complex, builtins.str, builtins.bytes]]
]

class _SupportsArray(Protocol[_DType_co]):
    def __array__(self) -> ndarray[Any, _DType_co]: ...
```

According to this type signature, `None` is not a valid member of `ArrayLike`. Therefore, as fas as I'm concerned, it is up to us what we want to do in this situation.

Here we are passing `None` in place of an array, for which we're asking NumPy to find the indices of non-zero elements. NumPy seems to have a built-in behavior where `None` gets converted to a a singleton `object`-dtype array which contains the object `None` (the definition of `ArrayLike` does not seem to document this behavior). Given that we don't broadly support `object`-dtype arrays, I don't think we want to replicate this behavior. I would rather we fail with an informative error message.

I think a good blanket solution for unexpected `None` inputs might be to add some error checking to `@add_boilerplate`, to throw a `ValueError` when `None` is passed where something `cunumeric.ndarray`-like is expected.

Reply:
> I would rather we fail with an informative error message.

OK well then let's leave the `xfail` tests for now then, since the current message is not very helpful